services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java

```diff
@@ -0,0 +1,535 @@
+
+        // set neighborInfos
+        List<NeighborInfo> neighborINFO = new ArrayList<>();
+        if (hasNeighbor) {
+            for (int i = 0; i < neighborNum; i ++) {
+                NeighborInfo neighborInfo = new NeighborInfo(DPMAutoUnitTestConstant.hostIp + i, DPMAutoUnitTestConstant.hostId, DPMAutoUnitTestConstant.portId + i, DPMAutoUnitTestConstant.portMac, DPMAutoUnitTestConstant.portIp, DPMAutoUnitTestConstant.vpcId, DPMAutoUnitTestConstant.subnetId + i);
+                neighborINFO.add(neighborInfo);
+            }
+        }
+        networkConfiguration.setNeighborInfos(neighborINFO);
+
+        // set neighborTable
+        List<NeighborEntry> neighborTable = new ArrayList<>();
+        for (int i = 0; i < L2IPNumInNeighborTable; i ++) {
+            int num1 = i + 2;
+            for (int j = i + 1; j < L2IPNumInNeighborTable; j ++) {
+                int num2 = j + 2;
+                NeighborEntry neighborEntry = new NeighborEntry();
+                neighborEntry.setLocalIp(DPMAutoUnitTestConstant.L2localIp + num1);
+                neighborEntry.setNeighborIp(DPMAutoUnitTestConstant.L2neighborIp + num2);
+                neighborEntry.setNeighborType(NeighborEntry.NeighborType.L2);
+                neighborTable.add(neighborEntry);
+            }
+        }
+
+        for (int i = 0; i < L3IPNumInNeighborTable; i ++) {
+            int num1 = i + 2;
+            for (int j = i + 1; j < L3IPNumInNeighborTable; j ++) {
+                int num2 = j + 2;
+                NeighborEntry neighborEntry = new NeighborEntry();
+                neighborEntry.setLocalIp(DPMAutoUnitTestConstant.L3localIp + num1);
+                neighborEntry.setNeighborIp(DPMAutoUnitTestConstant.L3neighborIp + num2);
+                neighborEntry.setNeighborType(NeighborEntry.NeighborType.L2);
+                neighborTable.add(neighborEntry);
+            }
+        }
+
+        for (int i = 0; i < L2IPNumInNeighborTable; i ++) {
+            int num1 = i + 2;
+            for (int j = 0; j < L3IPNumInNeighborTable; j ++) {
+                int num2 = j + 2;
+                NeighborEntry neighborEntry = new NeighborEntry();
+                neighborEntry.setLocalIp(DPMAutoUnitTestConstant.L2localIp + num1);
+                neighborEntry.setNeighborIp(DPMAutoUnitTestConstant.L3neighborIp + num2);
+                neighborEntry.setNeighborType(NeighborEntry.NeighborType.L3);
+                neighborTable.add(neighborEntry);
+            }
+        }
+        networkConfiguration.setNeighborTable(neighborTable);
+
+        // set portEntities
+        List<InternalPortEntity> portEntities = new ArrayList<>();
+        for (int j = 0; j < hostNum; j ++) {
+            for (int i = 0; i < portNum; i ++) {
+                List<PortEntity.FixedIp> fixedIps = new ArrayList<>();
+                PortEntity.FixedIp fixedIp = new PortEntity.FixedIp(DPMAutoUnitTestConstant.subnetId + i, DPMAutoUnitTestConstant.IpAddress + i);
```

Comment:
This will make each port fall into a different subnet. Don't think it is needed. Let us have a quick sync up. @kevin-zhonghao